wordpress-mobile/WordPress-iOS
# Jetpack Focus: Enable "What Is New" for Jetpack app

Part of #19164 

## Description

This PR enables the "What Is New" feature announcements for the Jetpack app.

- Updated `AppConfiguration.showsWhatIsNew` to `true` for Jetpack
- Extracted the `What's New in <app name>` string to `AppConstants`
- Extracted the `mobileAnnounceId` string to `AppConstants`

## How to test

Please test on both WPiOS and JPiOS.

1. Build and run the app
2. Go to the General settings for your app target (either WordPress or Jetpack)
3. Manually update the app version to `79.0` (a future / placeholder version for our test feature announcement)
4. Build and run the app again
5. ✅ Verify: the feature announcement should be displayed
6. Go to My Site > Me > App Settings
7. ✅ Verify: the` What's New in <app name>` row should be displayed
8. Tap on the `What's New in <app name>` row
9. ✅ Verify: the feature announcement should be displayed 

Test announcement | App settings
-- | --
<img src="https://user-images.githubusercontent.com/6711616/184623401-8188c1dc-cf6e-4f01-91e8-d2b1770e6025.png" width=200> | <img src="https://user-images.githubusercontent.com/6711616/184623406-7707bb7a-c1f7-409b-80fd-98d269e3bc09.png" width=200>
<img src="https://user-images.githubusercontent.com/6711616/184623471-e0e19543-6a10-4688-9c2d-d06b2ceeda80.png" width=200> | <img src="https://user-images.githubusercontent.com/6711616/184623475-a3db2d0d-dd8e-48a7-91df-6ddd045750ac.png" width=200>



## Regression Notes
1. Potential unintended areas of impact
- What's new on WP

4. What I did to test those areas of impact (or what existing automated tests I relied on)
- Tested the steps above

11. What automated tests I added (or what prevented me from doing so)
- n/a

PR submission checklist:
